lib/portlayer/exec/container.go

```diff
@@ -563,7 +616,9 @@ func (c *Container) LogReader(ctx context.Context, tail int, follow bool) (io.Re
 		}
 	}
 
-	if follow && c.State == StateRunning {
+	c.m.Lock()
```

Comment:
The lock should really be at the beginning of this function since we are accessing `c.vm`.